File: src/097/p097.go
package main

import (
	"fmt"
	"math/big"
)

func solve() int64 {
	num := big.NewInt(2)
	multiplier := big.NewInt(28433)
	exp := big.NewInt(7830457)
	mod := big.NewInt(1e10)

	prime := num.Exp(num, exp, mod).Mul(num, multiplier)

	one := big.NewInt(1)
	prime = prime.Add(prime, one)

	return prime.Int64() % 1e10
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/097/p097.java
import java.math.BigInteger;

public final class p097 {
    private static BigInteger solve() {
        BigInteger multiplier = BigInteger.valueOf(2).pow(7_830_457);
        BigInteger mod = BigInteger.valueOf(10_000_000_000L);

        return BigInteger.valueOf(28_433)
                .multiply(multiplier)
                .add(BigInteger.ONE)
                .remainder(mod);
    }

    public static void main(String[] args) {
        BigInteger result = solve();
        System.out.println(result);
    }
}
